rs/ingress_manager/src/ingress_handler.rs

```diff
@@ -142,6 +112,70 @@ impl<T: IngressPool> ChangeSetProducer<T> for IngressManager {
     }
 }
 
+enum IngressMessageValidationError {
+    IngressMessageTooLarge { max: usize, actual: usize },
+    UnexpectedStatus(IngressStatus),
+    InvalidRequest(RequestValidationError),
+}
+
+impl std::fmt::Display for IngressMessageValidationError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            IngressMessageValidationError::IngressMessageTooLarge { max, actual } => {
+                write!(f, "Ingress Message is too large {} > {}", actual, max)
+            }
+            IngressMessageValidationError::UnexpectedStatus(status) => write!(
+                f,
+                "Ingress Message is not `Unknown` to the IngressHistoryReader: {:?}",
+                status
+            ),
+            IngressMessageValidationError::InvalidRequest(error) => {
+                write!(f, "Ingress Message failed validation: {}", error)
+            }
+        }
+    }
+}
+
+impl IngressManager {
+    fn validate_ingress_pool_object(
+        &self,
+        ingress_object: &IngressPoolObject,
+        settings: &IngressMessageSettings,
+        ingress_message_status: impl Fn(&MessageId) -> IngressStatus,
+        consensus_time: Time,
+        registry_version: RegistryVersion,
+    ) -> Result<(), IngressMessageValidationError> {
+        // If the message is too large, consider the ingress message invalid
+        if ingress_object.signed_ingress.count_bytes() > settings.max_ingress_bytes_per_message {
```

Comment:
I think previously we compared against `ingress_object.count_bytes()`. Does it make a difference?

Reply:
hmmm, I'm not sure why I changed it while moving the code. It doesn't make any difference (https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/interfaces/src/ingress_pool.rs?L26-42) but I reverted it anyways. Thanks for catching it